samples/snippets/subscriber.py

```diff
@@ -1003,6 +1003,49 @@ def callback(message: pubsub_v1.subscriber.message.Message) -> None:
     # [END pubsub_dead_letter_delivery_attempt]
 
 
+def receive_messages_with_concurrency_control(
+    project_id: str, subscription_id: str, timeout: Optional[float] = None
+) -> None:
+    # [START pubsub_subscriber_concurrency_control]
+    from concurrent import futures
+    from google.cloud import pubsub_v1
+
+    # TODO(developer)
+    # project_id = "your-project-id"
+    # subscription_id = "your-subscription-id"
+    # Number of seconds the subscriber should listen for messages
+    # timeout = 5.0
+
+    # An optional executor to use. If not specified, a default one with maximum 10
+    # threads will be created.
+    executor = futures.ThreadPoolExecutor(max_workers=5)
+    # A thread pool-based scheduler. It must not be shared across SubscriberClients.
```

Comment:
Why can't the scheduler be shared across the `SubscriberClient`s? I thought they just ran tasks, oblivious to the actual logic. Does it have to do with cancellation?

Reply:
I didn't go deep into this and just copied the comment from [here](https://github.com/googleapis/python-pubsub/blob/main/google/cloud/pubsub_v1/subscriber/scheduler.py#L90-L91)